# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1533144926 14400
#      Wed Aug 01 13:35:26 2018 -0400
# Branch stable
# Node ID 216321ca4cc42aea1cc421ebc3422fc144d8eac4
# Parent  5625b2237a4d56c06a5da002797f7af736fb4cad
revive setstr (bug #54417)

* scripts/deprecated/setstr.m: Restore deleted function.

* scripts/deprecated/module.mk: Update.

diff --git a/scripts/deprecated/module.mk b/scripts/deprecated/module.mk
--- a/scripts/deprecated/module.mk
+++ b/scripts/deprecated/module.mk
@@ -8,16 +8,17 @@ FCN_FILE_DIRS += scripts/deprecated
   %reldir%/isstr.m \
   %reldir%/java2mat.m \
   %reldir%/mahalanobis.m \
   %reldir%/md5sum.m \
   %reldir%/octave_config_info.m \
   %reldir%/onenormest.m \
   %reldir%/paren.m \
   %reldir%/semicolon.m \
+  %reldir%/setstr.m \
   %reldir%/sleep.m \
   %reldir%/tmpnam.m \
   %reldir%/toascii.m \
   %reldir%/usleep.m \
   %reldir%/wavread.m \
   %reldir%/wavwrite.m
 
 %canon_reldir%dir = $(fcnfiledir)/deprecated
diff --git a/scripts/deprecated/setstr.m b/scripts/deprecated/setstr.m
new file mode 100644
--- /dev/null
+++ b/scripts/deprecated/setstr.m
@@ -0,0 +1,40 @@
+## Copyright (C) 2003-2018 John W. Eaton
+##
+## This file is part of Octave.
+##
+## Octave is free software; you can redistribute it and/or modify it
+## under the terms of the GNU General Public License as published by
+## the Free Software Foundation; either version 3 of the License, or (at
+## your option) any later version.
+##
+## Octave is distributed in the hope that it will be useful, but
+## WITHOUT ANY WARRANTY; without even the implied warranty of
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+## General Public License for more details.
+##
+## You should have received a copy of the GNU General Public License
+## along with Octave; see the file COPYING.  If not, see
+## <http://www.gnu.org/licenses/>.
+
+## -*- texinfo -*-
+## @deftypefn {Function File} {} setstr (@var{s})
+## This function has been deprecated.  Use char instead.
+## @end deftypefn
+
+## Author: jwe
+
+## Deprecated in version 3.0
+## Matlab still has this function, so don't remove just yet.
+
+function retval = setstr (varargin)
+
+  persistent warned = false;
+  if (! warned)
+    warned = true;
+    warning ("Octave:deprecated-function",
+             "setstr is obsolete and will be removed from a future version of Octave; please use char instead");
+  endif
+
+  retval = char (varargin{:});
+
+endfunction
